### DISCUSSION TITLE
Add search text inside folder

### DISCUSSION THREAD

**Discussion by thuan1412** [nvim-tree/nvim-tree.lua#1167] [Show and tell]
Hi, I have an idea about a new action for `nvim-tree`. That is search/find text in a folder. To accomplish this action, we may call a command from other plugins like `Telescope live_grep cwd={dir}}`. 

This action will be very useful when we work in a large project, where we just want to find the text from a single folder, not the whole project. Currently, I have to type a command to do this.

If you think this `action` is out of the scope of this plugin, maybe you consider supporting custom action, so that we can easily add a new action when working with nvim-tree.

**Comment by kyazdani42** [comment]
i think this could be very easily done in your config. You should just add a custom mapping, the node is passed as a parameter so you can send its path into the telescope command and then calling nvim tree find file. This should do the behavior you want.

**Reply by thuan1412** [reply]
Thank for your suggestion, I will try to implement it.

**Comment by thuan1412** [comment]
Here is my config to add this action.
```lua
local function getDirFromPath(path)
  if path:match("/") then
    return path:match("(.*)/")
  else
    return path:match("(.*)\\")
  end
end

local function findInFolder(node)
  local absolute_path = node.absolute_path
  local dir = getDirFromPath(absolute_path)
  require("telescope.builtin").live_grep({ cwd = dir })
end
....
    mappings = {
      custom_only = true,
      list = {
        { key = "f", action = "find in folder", action_cb = findInFolder },
      },
    },
....
```
